lms/views/api/canvas_studio.py

```diff
@@ -118,3 +119,30 @@ def list_collection(request):
     svc = request.find_service(CanvasStudioService)
     collection_id = request.matchdict["collection_id"]
     return svc.list_collection(collection_id)
+
+
+@view_config(
+    request_method="GET",
+    route_name="canvas_studio_api.via_url",
+    renderer="json",
+    permission=Permissions.API,
+)
+def via_url(request):
+    svc = request.find_service(CanvasStudioService)
+    document_url = request.params.get("document_url")
+    media_id = (
+        CanvasStudioService.media_id_from_url(document_url) if document_url else None
+    )
+    if not media_id:
+        raise HTTPBadRequest("Missing or invalid `document_url` param")
+
+    canonical_url = svc.get_canonical_video_url(media_id)
+    download_url = svc.get_video_download_url(media_id)
+    transcript_url = svc.get_transcript_url(media_id)
```

Comment:
This currently fetches the video download URL and transcript URL in two sequential API calls. It would be faster for the user if we could use two concurrent async calls.